chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/JHttpQuery.java

```diff
@@ -0,0 +1,372 @@
+package com.griddynamics.jagger.invoker.http.v2;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * An object that represents HTTP-request. It consists of {@link JHttpQuery#method},
+ * {@link JHttpQuery#headers}, {@link JHttpQuery#body}, {@link JHttpQuery#queryParams} and {@link JHttpQuery#clientParams} fields.
+ * <p>
+ * It contains methods which makes query construction concise and easy to read.
+ *
+ * @param <T> type of the query {@link JHttpQuery#body}
+ * @author Anton Antonenko
+ * @apiNote Example of query construction:
+ * <pre>{@code
+ * HashMap<String, String> cookies = new HashMap<>();
+ * cookies.put("JSESSIONID", "0123456789");
+ *
+ * HttpHeaders headers = new HttpHeaders();
+ * headers.add("header", "value");
+ *
+ * JHttpQuery<Integer> httpQuery = new JHttpQuery<Integer>()
+ *    .get()
+ *    .cookies(cookies)
+ *    .cookie("name", "value")
+ *    .headers(headers)
+ *    .header("header", newArrayList("value1", "value2"))
+ *    .queryParam("query param", "value")
+ *    .clientParam("client param", new Object())
+ *    .body(42);
+ * }</pre>
+ * @since 1.3
+ */
+@SuppressWarnings("unused")
+public class JHttpQuery<T> implements Serializable {
```

Comment:
Do you mean static inner Builder? If so, it cannot be used here, since this class is generalized and static class cannot be generalized.
Also, this kind of Builder is also valid.

Look at https://en.wikipedia.org/wiki/Method_chaining
and here is an example of this pattern
http://springfox.github.io/springfox/javadoc/current/springfox/documentation/spring/web/plugins/Docket.html